src/prometheus-net.DotNetRuntime/EventListening/Parsers/KestrelEventParser.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.Diagnostics.Tracing;
+using Prometheus.DotNetRuntime.EventListening.EventSources;
+using Prometheus.DotNetRuntime.EventListening.Parsers.Util;
+
+namespace Prometheus.DotNetRuntime.EventListening.Parsers
+{
+    public class KestrelEventParser : IEventParser<KestrelEventParser>, KestrelEventParser.Events.Info
```

Comment:
I can see you went through a fair bit of effort to write this class (the event parsers are by _far_ the most tedious part to write) so firstly, thankyou- you did a good job.

However, it seems that most if not all of the information you're pulling out here could be extracted using event counters. Event counters are definitely prefered for any event that is going to be high-frequency and pretty much required on .NET 3.1 to avoid a nasty CPU leak.

It seems asp.net core 3.1 has the following event counter sources available to it:
- [Hosting event source](https://github.com/dotnet/aspnetcore/blob/52eff90fbcfca39b7eb58baad597df6a99a542b0/src/Hosting/Hosting/src/Internal/HostingEventSource.cs)
- [Kestrel event source](https://github.com/dotnet/aspnetcore/blob/350cf6bd715e55b4da88ad2e9a843255dd547a5c/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelEventSource.cs)
- [ConcurrencyLimiterEventSource](https://github.com/dotnet/aspnetcore/blob/52eff90fbcfca39b7eb58baad597df6a99a542b0/src/Middleware/ConcurrencyLimiter/src/ConcurrencyLimiterEventSource.cs)

`OnEventCommand` is the method to look at in each class- it will initialize the counters and specifying the name of the counter. Using this name we can subscribe to counters (see `RuntimeEventParser` as an example).